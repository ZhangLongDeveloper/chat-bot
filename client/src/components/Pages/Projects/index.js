import React from 'react';

import EmptyView from '../../Common/EmptyView';

const Projects = () => (
  <div>
    <EmptyView section="projects" />
  </div>
);

export default Projects;
