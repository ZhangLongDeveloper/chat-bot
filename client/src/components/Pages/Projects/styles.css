.projects-section-container {
  padding-bottom: 50px;
}
.projects-section-container .projects-section-title {
  text-align: center;
  margin: 35px 0;
}
.projects-section-container .projects-wrapper {
  margin-bottom: 50px;
}

@media (max-width: 992px) {
  .projects-section-container .projects-section-title {
    margin: 25px 0 10px;
  }
}

@media (max-width: 600px) {
  .projects-section-container .projects-section-title {
    font-size: 32px;
    margin: 25px 0 0 0;
  }
}