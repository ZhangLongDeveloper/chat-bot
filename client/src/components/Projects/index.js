import React from 'react';

const Projects = () => (
  <div>
    This is PROJECTS
  </div>
);

export default Projects;
