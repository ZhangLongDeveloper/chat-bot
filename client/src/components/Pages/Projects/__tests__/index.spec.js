import React from 'react';
import { create } from 'react-test-renderer';

import Projects from '../../Projects';

jest.mock('../../../Common/EmptyView', () => 'EmptyView');

const mockComponent = props => (
  <Projects {...props} />
);

describe('components/pages/<Projects />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
