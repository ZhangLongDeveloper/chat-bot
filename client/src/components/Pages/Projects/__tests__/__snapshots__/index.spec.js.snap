// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pages/<Projects /> should render component 1`] = `
<div
  className="projects-section-container"
>
  <h3
    className="projects-section-title"
  >
    Projects
  </h3>
  <div
    className="projects-wrapper"
  >
    <ProjectItem
      description="
    This is a simple blog about traveling written with MERN stack.
    The purpose of the project is to learn and practice a full
    stack creation of the app! This is my first project where I
    implemented a custom authentication system, mongo CRUD, configured
    a CI with linting and 100% client-side test coverage. As in my
    previous projects, I used a very strict git flow, detailed
    documentation of the development, versioning and descriptive
    releases, automated Kanban projects and issues to organize the process.
    Also, this project has two environments (PROD and PRE) for
    testing and integration purposes. I realize that this project
    is not yet perfect but it is the most professional project I have
    implemented for the moment (from both code and organization perspective).
  "
      id={4}
      image="test-file-stub"
      name="TripReviewer"
      tools={
        Array [
          "React",
          "Redux",
          "Jest",
          "Styled Components",
          "Node",
          "Express",
          "MongoDB",
          "JWT",
          "Firebase",
        ]
      }
      url="https://github.com/KovDimaY/Trip-Reviewer"
    />
    <ProjectItem
      description="
    This is a simple chat with a support of different rooms
    written with NodeJS and Socket.IO. The purpose of the project
    is to learn and practice how to use Socket.IO and to have fun!
    This is my first project where I incorporated \\"Projects\\" feature
    from GitHub to manage the process of development with an automated
    Kanban board and also this is my first project where I use Versioning
    and descriptive Releases. This project has a strict development
    organization with 3-layer git flow (master, development and feature
    branches) and all the features and development process are well
    documented using Github Issues and descriptive commits. Just
    tried to make it well organized and to feel like a real company
    development process.
  "
      id={3}
      image="test-file-stub"
      name="SimpleChat"
      tools={
        Array [
          "Node",
          "jQuery",
          "Socket.io",
          "Mocha",
        ]
      }
      url="https://github.com/KovDimaY/SimpleChat-Socket.io"
    />
    <ProjectItem
      description="
    This project was created to learn and practice my
    knowledge of React.js and to integrate Highcharts as
    react components in an application with Express.js back-end.
    Also, I tried to collect in one place all my knowledge
    about Highcharts and how to work with them, where it
    will be not only useful but also interesting to see results.
    This project can be useful for people who are not very
    familiar with the Highcharts and who want to see different
    applications and use-cases of all the basic charts, using
    React to manage all the interactions.
  "
      id={2}
      image="test-file-stub"
      name="React-Highcharts Playground"
      tools={
        Array [
          "React",
          "Highcharts",
          "Webpack",
        ]
      }
      url="https://github.com/KovDimaY/React-Highcharts"
    />
    <ProjectItem
      description="
    This is a classical Memory Game created with pure
    Javascript to practice DOM manipulations and Local Storage usage.
    There is a couple of options that player can use to customize this game.
    User has a possibility to choose the number of elements to
    play with (12, 18, 24 or 30) and the type of the elements.
    Currently there are next game modes: numbers, characters,
    colors and pictures.
  "
      id={5}
      image="test-file-stub"
      name="Memory Game"
      tools={
        Array [
          "HTML",
          "CSS",
          "JavaScript",
        ]
      }
      url="https://github.com/KovDimaY/Memory-Game"
    />
    <ProjectItem
      description="
    This project was created to practice my skills in programming,
    game design and game development in general, and also to satisfy
    my passion to create. There are several games developed using Microsoft
    XNA Framework for C# and Unity. All games have documentation such as
    Game Design Document and Pitch Document.
    Games are free to download, and they can be played on Windows
    computers. Hope people will enjoy playing them.
  "
      id={1}
      image="test-file-stub"
      name="KovDimaY"
      tools={
        Array [
          "C#",
          "XNA Framework",
          "Unity",
        ]
      }
      url="http://kovalenkody.wixsite.com/kovdimay"
    />
  </div>
</div>
`;
